### PR TITLE
feat: language provider (windows only)

### PR DESCRIPTION
**Pull request by jawee**
Provider for getting current OS language and keyboard language.

Only Windows support, developed the same thing for the c# version of glazewm.

Haven't been able to run it, `pnpm dev` fails with the error below. So it's untested. 
```
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @zebar/desktop@0.0.0 dev: `npm run -s monitors -- --print0 | xargs -0 -P 99 
-I % sh -c 'npm run tauri dev -- -- -- open bar --args %'`
Exit status 255
 ELIFECYCLE  Command failed with exit code 255.
```

### CODE REVIEW

**Comment by lars-berger** [comment]
Ay this is awesome, thank you 🙏 

A big improvement that could be made is removing the interval altogether and instead listening to `WM_INPUTLANGCHANGE`. This way we wouldn't have delays in showing the current language. If you're ok with making the change, [`create_message_window`](https://github.com/glzr-io/glazewm/blob/main/packages/wm/src/common/platform/platform.rs#L161) and [`run_message_loop`](https://github.com/glzr-io/glazewm/blob/main/packages/wm/src/common/platform/platform.rs#L204) can be copied 1:1 from GlazeWM. You'd need to spawn a thread for the window [like so](https://github.com/glzr-io/glazewm/blob/main/packages/wm/src/common/platform/event_window.rs#L92) and have an [`event_hook_proc`](https://github.com/glzr-io/glazewm/blob/main/packages/wm/src/common/platform/event_window.rs#L170) that listens for `WM_INPUTLANGCHANGE`

Re. `pnpm dev` failing - are you able to run `main` without the pr changes?

File: packages/desktop/src/providers/manager.rs [hunk 1]
```diff
@@ -0,0 +1,305 @@
+use std::{
```
Comment (lars-berger) [inline packages/desktop/src/providers/manager.rs:1]: file should be renamed to `provider_manager.rs` - might be the source of the compile error

File: packages/desktop/src/providers/variables.rs [hunk 2]
```diff
@@ -20,4 +20,6 @@ pub enum ProviderVariables {
   Memory(MemoryVariables),
   Network(NetworkVariables),
   Weather(WeatherVariables),
+  #[cfg(all(windows, target_arch = "x86_64"))]
```
Comment (lars-berger) [inline packages/desktop/src/providers/variables.rs:23]: should work fine on arm64 (and we don't compile for 32bit), so can be simplified to just `#[cfg(windows)]`

File: packages/client-api/src/providers/language/create-language-provider.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export interface LanguageVariables {
+  language: string;
+}
+
+export async function createLanguageProvider(
+  config: LanguageProviderConfig,
+  owner: Owner,
+) {
+  const ipVariables = await createProviderListener<
```
Comment (lars-berger) [inline packages/client-api/src/providers/language/create-language-provider.ts:14]: typo

**Comment by lars-berger** [comment]
> Re pnpm dev, it was a bit of a user error (guess I can't read above headlines :) ), just noticed that git bash or wsl was required, and I was using powershell. Now it kind of works, although no providers seems to work, getting the error below for all of them (I assume all of them at least), can't get any of the default providers in "group/right" to work. This also happens if I try to run it from main. `WARN zebar::providers::provider_manager: Error refreshing provider: channel closed`
> 
> Want me to move the "problem discussion" to an issue, to not clutter the PR further? :)

Ah thanks for catching this - was a regression introduced by a recent refactor. There's a fix in `main` for it now 👍

**Comment by jawee** [comment]
It does seem like the WM_INPUTLANGCHANGE only is being sent to the focused window and then it's being swallowed. If I create a visible window and do a language change, the message is received. But regardless of what I seem to do, I can't get it to be sent to a background window. So not sure how to proceed, other than cleaning up my polling version or wait for someone that might know the windows api intricacies better than me to figure it out.

**Comment by jawee** [comment]
I've tried implementing it using the ITfLanguageProfileNotifySink, but haven't been able to get it to work with zebar, I did make a small application where I've gotten it to work, but it seems to have the same issue as the WM_INPUTLANGCHANGE event, it's only sent to the application in focus. 

According to this, it seems like that's how it's supposed to be. https://stackoverflow.com/questions/74468291/itflanguageprofilenotifysinkonlanguagechange-only-reporting-when-the-window-is

Doesn't seem like this feature can be done in a nice way unfortunately.

**Pull request closed without merging**